commcare_connect/opportunity/views.py

```diff
@@ -494,6 +494,9 @@ def add_payment_unit(request, org_slug=None, pk=None):
             parent_payment_unit=form.instance.id
         )
         messages.success(request, f"Payment unit {form.instance.name} created.")
+        claims = OpportunityClaim.objects.filter(opportunity_access__opportunity=opportunity)
+        for claim in claims:
+            OpportunityClaimLimit.create_claim_limits(opportunity, claim)
```

Comment:
Nice, is it this simple a fix. Have you verified the budget/limit consistency among users/payment_units?

Reply:
looks good pending the answer to this, and a test would be great

Reply:
Yeah, this is currently deployed on staging and the budget was updated correctly when a new payment unit was added.